frontend/public/components/monitoring/dashboards/index.tsx

```diff
@@ -690,6 +742,22 @@ const MonitoringDashboardsPage: React.FC<MonitoringDashboardsPageProps> = ({ mat
         return acc;
       }, []);
 
+  const handleDashboardChange = (val: string) => {
```

Comment:
Can we use something more descriptive than `val`. Maybe `newBoard`?

Reply:
I think this `handleDashboardChange` function should be merged into the `useEffect` above. Otherwise, it's fairly confusing with both of them changing the URL. In particular, `handleDashboardChange` calls `changeBoard`, which will then trigger the `useEffect`, so it's fairly hard to reason about.